### PR TITLE
fix: fix hydration issue

### LINKED ISSUE #6129: ChakraProvider breaks Suspense when reloading in Next.js project
### Description

When I reloaded the screen I used with react 18 suspense and Next.js, I expected to display correctly happen but instead an error was displayed.

### Link to Reproduction

https://codesandbox.io/s/stoic-violet-b8beh4?file=/pages/index.tsx

### Steps to reproduce

1. Create Next.js App
2. Add a Suspense component into pages/index.tsx
3. Verify it works fine (should)
4. Add @chakra-ui/react and <ChakraProvider> into _app.tsx
5. Reload the page - emits the following error:

```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.
```

### Chakra UI Version

2.1.2

### Browser

Firefox, Chrome, Safari

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

This issue seemed to have solved the problem, but it still seems to be occurring.
Please check.

https://github.com/chakra-ui/chakra-ui/issues/5842

### PR DESCRIPTION
**Pull request by itkrt2y**
Closes #6129

## 📝 Description

Fix hydration errors caused by `EnvironmentProvider`, `Portal` and `useMediaQuery` in SSR + Suspense.

## ⛳️ Current behavior (updates)

Not work `EnvironmentProvider`, `Portal` and `useMediaQuery` in SSR + Suspense

## 🚀 New behavior

Works as expected

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Comment by itkrt2y** [comment]
1. I created a testable next.js app. Run `yarn dev` in `examples/next-url/` to start the app.
    - https://github.com/itkrt2y/chakra-ui/tree/fix-hydration-test
2. The sandbox automatically generated in this PR is still broken, but I think this is a bug in the sample code and urql, not chakra. 
    - Maybe the sample code bug can be fixed with the `next-urql` package

**Comment by itkrt2y** [comment]
@segunadebayo Thanks for review! I added changeset. 3d8e095dfc696b3d903455319231e77d1c21d875

**Review by segunadebayo — COMMENTED** [review]
Thanks for looking into this @itkrt2y. See my comment above

File: packages/media-query/src/use-media-query.ts [hunk 1]
```diff
@@ -37,17 +40,18 @@ export function useMediaQuery(
...
-      )
-    }
+    setValue(
+      queries.map((query) => ({
+        media: query,
+        matches: env.window.matchMedia(query).matches,
+      })),
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```
Comment (segunadebayo) [inline packages/media-query/src/use-media-query.ts:52]:
```suggestion
  }, [env.window])
```

We probably want to include the `env.window` in the deps

File: packages/media-query/src/use-media-query.ts [hunk 2]
```diff
@@ -37,17 +40,18 @@ export function useMediaQuery(
...
 
+  useSafeLayoutEffect(() => {
```
Comment (segunadebayo) [inline packages/media-query/src/use-media-query.ts:54]: I think it's okay to move this code into the `useEffect` for simplicity.

Comment (itkrt2y) [reply packages/media-query/src/use-media-query.ts]: refactored be45b1a5100079ed78451652a0f038f716a82680

**Review by segunadebayo — APPROVED** [review]
Looks good! Thanks for walking through the reviews and changes with me. I appreciate it.

**Pull request merged**